src/allmydata/test/common_util.py

```diff
@@ -8,12 +8,10 @@
 from twisted.trial import unittest
 
 from ..util.assertutil import precondition
+from ..scripts import runner
 from allmydata.util.encodingutil import get_io_encoding
-from future.utils import PY2
-if PY2: # XXX this is a hack that makes some tests pass on Python3, remove
-        # in the future
-    from ..scripts import runner
 # Imported for backwards compatibility:
+from future.utils import bord, bchr, binary_type
 from .common_py3 import (
     SignalMixin, skip_if_cannot_represent_filename, ReallyEqualMixin, ShouldFailMixin
 )
```

Comment:
I think it is the `common_py3` imports that are being described as "for backwards compatibility" - specifically, for API compatibility so that even though they're implemented in a separate, ported module they can still be imported via their original names.

Reply:
Yes, but then aren't we also importing from `future.utils` for backwards compatibility, in a sense? That's how I was thinking about it.